src/Microsoft.AspNet.Mvc.Core/SessionStateTempDataProvider.cs

```diff
@@ -61,11 +61,16 @@ public class SessionStateTempDataProvider : ITempDataProvider
                 return null;
             }
 
-            var tempDataDictionary = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
             var session = context.Session;
+            if (session == null)
+            {
+                return null;
```

Comment:
Returning `null` here changes the behavior https://github.com/aspnet/Mvc/pull/2647/files#diff-66225f3019f3fc28323c68c616ec48b9R29. Is this intentional?